### DISCUSSION TITLE
400 Error, overspend account

### DISCUSSION THREAD

**Discussion by hmzdot** [reach-sh/reach-lang#936] [Q&A] [answered]
Xxiled#0317's question

> Hi all - quick one (hopefully) - any reason i'd be getting a 400, overspend error on my first use of: 
> `acc = await stdlib.newTestAccount(startingBalance);`?

**Comment by hmzdot** [comment, marked as answer]
This generally happens when standard library version doesn't match with the version of reach compiler contracts are compiled with

Reasoning:
When a new test account is created it is funded with a starting balance from an account called "faucet". 
This account's address changes with each version so whenever there's a new update, trying to take funds from a previous one which has no tokens results with an `overspend account` error

Mitigation:
- Stop old instances of reach: `./reach down; ./reach docker-reset`
- Update reach compiler: `./reach update`
- Recompile existing contracts: `./reach compile`
- Update your standard library on the frontend, if there's one: `npm update @reach-sh/stdlib`

**Reply by xxiled-plastic-cat** [reply]
Ah, that makes sense! Will give that a go, glad to hear it sounds relatively normal!

**Reply by xxiled-plastic-cat** [reply]
@zetsuboii so I've tried the above mitigation steps but to no avail - still getting the same error. I'm running reach version 84dc282c 0.1.10

**Reply by jeapostrophe** [reply]
Do you have a `package.json` file?

**Reply by xxiled-plastic-cat** [reply]
I do yeah:
{
  "name": "@reach-sh/bread",
  "type": "module",
  "dependencies": {
  },
  "author": "reach.sh",
  "license": "Apache-2.0",
  "scripts": {
    "lint": "eslint --ignore-path .gitignore --ext .mjs .",
    "index": "node --experimental-modules --unhandled-rejections=strict index.mjs"
  }
}

**Reply by jeapostrophe** [reply]
Do you have a package "lock" file?

**Reply by xxiled-plastic-cat** [reply]
<img width="397" alt="Screenshot 2022-05-09 at 20 48 13" src="https://user-images.githubusercontent.com/93330009/167486272-2caa1dd6-a7e9-4659-b791-2a86eb9e11ce.png">
Not that I can see, no

**Reply by jeapostrophe** [reply]
Okay, next thing to check is your Dockerfile it could be pinned to an old release

**Reply by xxiled-plastic-cat** [reply]
Dockerfile:

```
FROM reachsh/runner:0.1.7

# If your project needs more node dependencies:
# COPY package.json /app/package.json
# RUN npm install
# RUN npm link @reach-sh/stdlib

COPY . /app
CMD ["index"]
```

**Reply by jeapostrophe** [reply]
You have pinned it to 0.1.7 but are trying to use it with a 0.1.10 devnet

**Reply by xxiled-plastic-cat** [reply]
Ah, that's it! Thanks! What a stupid mistake!
